pkg/artifactory/datasource/repository/local/datasource_artifactory_local_docker_repository.go

```diff
@@ -0,0 +1,56 @@
+package local
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/jfrog/terraform-provider-artifactory/v6/pkg/artifactory/resource/repository/local"
+	"github.com/jfrog/terraform-provider-shared/packer"
+	"github.com/jfrog/terraform-provider-shared/util"
+)
+
+const packageType = "docker"
+
+func DataSourceArtifactoryLocalDockerV2Repository() *schema.Resource {
+	pkr := packer.Default(local.DockerV2LocalSchema)
+
+	constructor := func() (interface{}, error) {
+		return &local.DockerLocalRepositoryParams{
+			RepositoryBaseParams: local.RepositoryBaseParams{
+				PackageType: packageType,
+				Rclass:      rclass,
+			},
+			DockerApiVersion:    "V2",
+			TagRetention:        1,
+			MaxUniqueTags:       0, // no limit
+			BlockPushingSchema1: true,
+		}, nil
+	}
+
+	return &schema.Resource{
+		Schema:      local.DockerV2LocalSchema,
+		ReadContext: MkRepoReadDataSource(pkr, constructor),
+	}
+}
+
+func DataSourceArtifactoryLocalDockerV1Repository() *schema.Resource {
+	// this is necessary because of the pointers
+	skeema := util.MergeMaps(local.DockerV1LocalSchema)
```

Comment:
I wonder why this is necessary for this but for not the others?

Reply:
I'm not sure, its done in the regular resource so I copied it over here. I've run into this pointer issue before, and it didn't work without this for a more complex test in the resource files. I will see if its necessary for data sources, kind of assumed it was